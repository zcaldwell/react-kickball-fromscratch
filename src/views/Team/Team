import React from 'react';
import { fetchPlayers } from '../../services/players';

export default function Team() {
  return <div></div>;
}
