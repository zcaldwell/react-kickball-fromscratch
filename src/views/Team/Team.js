import React from 'react';
import { useState, useEffect } from 'react';
import { fetchTeamById } from '../../services/teams';
import TeamDetail from '../../components/TeamDetail';

export default function Team(props) {
  const [team, setTeam] = useState({ players: [] });

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchTeamById(props.match.params.id);
      console.log(data);
      setTeam(data[0]);
    };
    fetchData();
  }, [props]);

  return (
    <>
      <TeamDetail team={team} />
    </>
  );
}
