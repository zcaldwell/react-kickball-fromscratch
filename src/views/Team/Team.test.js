import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Team from './Team';

test('should render player details', async () => {
  const { container } = render(
    <MemoryRouter>
      <Team match={{ params: { id: 2 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Stumptown Lumberjacks');
  expect(container).toMatchSnapshot();
});
