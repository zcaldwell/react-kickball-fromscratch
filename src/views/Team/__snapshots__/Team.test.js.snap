// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render player details 1`] = `
<div>
  <div>
    <h1>
       
      Stumptown Lumberjacks
       
    </h1>
    <h2>
      Portland
      , 
      OR
    </h2>
    <li>
      <a
        href="/players/2"
      >
        Gary Coolman
      </a>
    </li>
    <li>
      <a
        href="/players/2"
      >
        Terry Cruise
      </a>
    </li>
    <li>
      <a
        href="/players/2"
      >
        Bud E. Guy
      </a>
    </li>
    <li>
      <a
        href="/players/2"
      >
        Buster Rimes
      </a>
    </li>
  </div>
</div>
`;
